rotkehlchen/tasks/manager.py

```diff
@@ -696,15 +700,40 @@ def _maybe_update_morpho_cache(self) -> Optional[list[gevent.Greenlet]]:
 
         return greenlets if len(greenlets) > 0 else None
 
+    def _maybe_update_pendle_cache(self) -> Optional[list[gevent.Greenlet]]:
+        with self.database.conn.read_ctx() as cursor:
+            account_data = self.database.get_blockchain_accounts(cursor)
+            if (
+                len(account_data.get(SupportedBlockchain.ARBITRUM_ONE)) == 0 and
+                len(account_data.get(SupportedBlockchain.ETHEREUM)) == 0 and
+                len(account_data.get(SupportedBlockchain.BASE)) == 0 and
+                len(account_data.get(SupportedBlockchain.BINANCE_SC)) == 0 and
+                len(account_data.get(SupportedBlockchain.OPTIMISM)) == 0
+            ):
+                return None
+
+        return [
+            self.greenlet_manager.spawn_and_track(
+                after_seconds=None,
+                task_name=f'Update Pendle yield tokens for {chain.to_name()}',
+                exception_is_error=False,
+                method=self.query_pendle_yield_tokens,
+                database=self.chains_aggregator.get_chain_manager(chain.to_blockchain()),  # type: ignore[arg-type]  # chain is supported
+            )
+            for chain in PENDLE_SUPPORTED_CHAINS_WITHOUT_ETHEREUM | {ChainID.ETHEREUM}
+            if should_update_protocol_cache(CacheType.PENDLE_YIELD_TOKENS, (str(chain.serialize()),)) is True  # noqa: E501
+        ]
+
     def _maybe_update_aura_pools(self) -> Optional[list[gevent.Greenlet]]:
         with self.database.conn.read_ctx() as cursor:
+            account_data = self.database.get_blockchain_accounts(cursor)
             if (
-                len(self.database.get_single_blockchain_addresses(cursor, SupportedBlockchain.ETHEREUM)) == 0 and  # noqa: E501
-                len(self.database.get_single_blockchain_addresses(cursor, SupportedBlockchain.BASE)) == 0 and  # noqa: E501
-                len(self.database.get_single_blockchain_addresses(cursor, SupportedBlockchain.OPTIMISM)) == 0 and  # noqa: E501
-                len(self.database.get_single_blockchain_addresses(cursor, SupportedBlockchain.POLYGON_POS)) == 0 and  # noqa: E501
-                len(self.database.get_single_blockchain_addresses(cursor, SupportedBlockchain.ARBITRUM_ONE)) == 0 and  # noqa: E501
-                len(self.database.get_single_blockchain_addresses(cursor, SupportedBlockchain.GNOSIS)) == 0  # noqa: E501
+                len(account_data.get(SupportedBlockchain.ETHEREUM)) == 0 and
+                len(account_data.get(SupportedBlockchain.BASE)) == 0 and
+                len(account_data.get(SupportedBlockchain.GNOSIS)) == 0 and
+                len(account_data.get(SupportedBlockchain.ARBITRUM_ONE)) == 0 and
```

Comment:
nitpick: here keep same order in the checks, otherwise it's easier to miss mistakes in the diff check